runtime/interpreter/interpreter_expression.go

```diff
@@ -212,6 +221,23 @@ func (interpreter *Interpreter) memberExpressionGetterSetter(memberExpression *a
 				}
 			}
 
+			// Return a reference, if the member is accessed via a reference.
```

Comment:
In the current implementation, the interpreter detects if a reference needs to be returned based on the value, on each evaluation. Would it be possible to determine this statically, i.e. move the logic here to the checker and store the result in the elaboration?

Reply:
Good idea. Added in 5c8ea343e51e568e3fd756f233c97436556c82af